#52 Login using Identity library

BirdyDK opened the pull request.

Emilbks reviewed (changes requested):
ada

CptShini reviewed (commented):
Looks very fine, quite epic.

However... I don't really understand the sudden change to using strings for AuthorId's everywhere in the code, which I'd love to know why we're doing lol.

Secondly, you changed the Environment variable code I wrote, and it's actually a bit faulty now :/ as the environment string can in fact be null, which I (for good reasons) test for.

BirdyDK commented:
Thanks for the feedback. I changed back the environment variable code.

The reason why I have changed all of the ID attributes and variables is that our Author domain is now inheriting from IdentityUser, which has Id as a string.

CptShini reviewed (approved):
Code seems good, adressed the problems I previously had, and registering/logging in works.

Cybermanboss commented on src/Chirp.Razor/Areas/Identity/Pages/Account/Login.cshtml:
why keep the outcommented code? is it for later use, or did you just forget to remove it?

BirdyDK replied:
It is for later use when we setup login with github

Cybermanboss commented on src/Chirp.Razor/Program.cs:
you missed the nullable notation for the string in line 73

Cybermanboss replied:
and line 65 and 88

Cybermanboss replied:
also github warns about unreachable code, although it is used right after, so just be sure that it works

BirdyDK replied:
The places you mentioned are marked as nullable and I assume the code for test enviroment works since the tests work.

Cybermanboss reviewed (approved):
there was some very small stuff, which might or might not change whether program.cs file works as intended. but as it runs i will approve
